src/functions/insult/main.py

```diff
@@ -0,0 +1,69 @@
+import json
+import re
+from typing import Any, Optional, Tuple
+
+import flask
+import functions_framework
+import requests
+
+REPLACE_USER_NAME = "!_A_USER_NAME_!"
+BASE_INSULT = "So bad even Archy can't stand you, jk <3"
+StrL = list[str]
+
+
+@functions_framework.http
+def insult(request: flask.Request) -> Tuple[str, int]:
+    """Return a random insult to the tagged user or the author"""
+
+    request_json: Optional[Any] = request.get_json(silent=True)
+
+    if request_json:
+        mentions: list = request_json.get("mentions", None)
+
+        if mentions is None or len(mentions) == 0:
+            response: requests.Response = requests.get("https://insult.mattbas.org/api/en/insult.json")
+            return get_the_insult(response.status_code, response.content), 200
```

Comment:
Just pass the full response object, don't split it in two

`return get_insult(response), 200`